From b7bec7e4b124afae16ba54406487ad7fefac4f98 Mon Sep 17 00:00:00 2001
From: Rameshkumar Sundaram <quic_ramess@quicinc.com>
Date: Sat, 4 Jun 2022 01:42:01 +0530
Subject: [PATCH] ath12k: Send Puncturing pattern of peer mesh during assoc

Mesh Peers can have different puncturing pattern than self
mesh vdev, send the pattern advertised by peer mesh STA in
peer assoc params.
FW will intersect Peer and self mesh vdev puncture atterns
and decide EHT PPDU Tx BW for the peer.

Signed-off-by: Rameshkumar Sundaram <quic_ramess@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index e733ae5..7959226 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -2518,10 +2518,12 @@ static void ath12k_peer_assoc_h_eht(struct ath12k *ar,
 
 	arg->peer_nss = min(sta->rx_nss, max_nss);
 	arg->ru_punct_bitmap = ~def.ru_punct_bitmap;
+	if (ieee80211_vif_is_mesh(vif) && sta->ru_punct_bitmap)
+		arg->ru_punct_bitmap = ~sta->ru_punct_bitmap;
 
 	ath12k_dbg(ar->ab, ATH12K_DBG_MAC,
-		   "mac he peer %pM nss %d mcs cnt %d\n",
-		   sta->addr, arg->peer_nss, arg->peer_he_mcs_count);
+		   "mac he peer %pM nss %d mcs cnt %d ru_punct_bitmap 0x%x\n",
+		   sta->addr, arg->peer_nss, arg->peer_he_mcs_count, arg->ru_punct_bitmap);
 }
 
 static void ath12k_peer_assoc_h_smps(struct ieee80211_sta *sta,
-- 
2.7.4

